### PR TITLE
Implemented Paper's Discriminator in Keras

### PR DESCRIPTION
**Pull request by lamdav**
See `model_factory.py` for raw implementation.
See `discriminator.ipynb` for a notebook on usage and more details.

### CODE REVIEW

File: src/discriminator/model_factory.py [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+            filters *= 2
+
+            intermediate_kernel_size = (3, 3)
+            if i < 2:
+                intermediate_kernel_size = (4, 4)
+            layer = intermediate_layer(
+                layer, filters, intermediate_kernel_size)
+
+        outputs = Dense(2, activation="sigmoid")(layer)
+        return Model(inputs=inputs, outputs=outputs)
```
Comment (zhoutwo) [inline src/discriminator/model_factory.py]: Please compile the model

File: src/discriminator/model_factory.py [hunk 2]
```diff
@@ -0,0 +1,118 @@
+from keras.models import Model
+from keras.layers import Input, Conv2D, BatchNormalization, Activation, LeakyReLU, Add, Dense
```
Comment (zhoutwo) [inline src/discriminator/model_factory.py]: BatchNormalization is not used in this code

**Review by zhoutwo — CHANGES_REQUESTED** [review]

File: src/discriminator/model_factory.py (hunk 2 above)
Comment (lamdav) [reply src/discriminator/model_factory.py]: fixed.

File: src/discriminator/model_factory.py (hunk 1 above)
Comment (lamdav) [reply src/discriminator/model_factory.py]: fixed

**Review by zhoutwo — APPROVED** [review]
LGTM
